src/graph/serializers/serialize_graph.c

```diff
@@ -48,6 +48,32 @@ static NodeID _updatedID(uint64_t *array, NodeID id) {
     }
 }
 
+/* Create a mapping from attribute ID to attribute name. */
+static char** _Schema_AttributeMapping(Schema *s, unsigned short *attr_count) {
+    *attr_count = Schema_AttributeCount(s);
+    char **attribute_map = malloc(sizeof(char*) * (*attr_count));
+    
+    char *ptr;
+    tm_len_t len;
+    Attribute_ID *attr_id;
+    TrieMapIterator *it = TrieMap_Iterate(s->attributes, "", 0);
+
+    while(TrieMapIterator_Next(it, &ptr, &len, (void**)&attr_id)) {
+        attribute_map[*attr_id] = strdup(ptr);
+        attribute_map[*attr_id][len] = '\0';
```

Comment:
I haven't seen this fail, but it doesn't seem safe to me? `strdup` internally behaves like `strlen` to find a NULL byte, which the triemap doesn't guarantee will be found (which is why we're adding the NULL byte afterwards here.

I might be missing something that guarantees the behavior in these lines. In general, though, I feel like we might be able to reduce the number of copies in schema deserialization if we encode NULL bytes into the trie strings, or at least into the RDB file.

Reply:
I definitely agree!